bellman/src/groth16/generator.rs

```diff
@@ -215,8 +215,22 @@ where
         assembly.num_inputs + assembly.num_aux
     });
 
-    let gamma_inverse = gamma.inverse().ok_or(SynthesisError::UnexpectedIdentity)?;
-    let delta_inverse = delta.inverse().ok_or(SynthesisError::UnexpectedIdentity)?;
+    let gamma_inverse = {
+        let inverse = gamma.invert();
```

Comment:
I think that makes it too easy to slip out of constant-time-ness, which could lead to subtle bugs. I'm rather partial to the fact that if you want to leave that domain, it needs to be noisy. It would definitely be nice to have it be easier than it currently is; it should be possible to do that while still being noisy.